packages/server/lib/cloud/artifacts/screenshot_artifact.ts

```diff
@@ -0,0 +1,44 @@
+import fs from 'fs/promises'
+import Debug from 'debug'
+import { Artifact, IArtifact } from './artifact'
+import { fileUploadStrategy } from './file_upload_strategy'
+
+const debug = Debug('cypress:server:cloud:artifacts:screenshot')
+
+const createScreenshotArtifact = async (filePath: string, uploadUrl: string): Promise<IArtifact | undefined> => {
+  try {
+    const { size } = await fs.stat(filePath)
+
+    return new Artifact('screenshots', filePath, uploadUrl, size, fileUploadStrategy)
+  } catch (e) {
+    debug('Error creating screenshot artifact: %O', e)
+
+    return
+  }
+}
+
+export const createScreenshotArtifactBatch = (
+  screenshotUploadUrls: {screenshotId: string, uploadUrl: string}[],
+  screenshotFiles: {screenshotId: string, path: string}[],
+): Promise<IArtifact[]> => {
+  const correlatedPaths = screenshotUploadUrls.map(({ screenshotId, uploadUrl }) => {
+    const correlatedFilePath = screenshotFiles.find((pathPair) => {
+      return pathPair.screenshotId === screenshotId
+    })?.path
+
+    return correlatedFilePath ? {
+      filePath: correlatedFilePath,
+      uploadUrl,
+    } : undefined
+  }).filter((pair): pair is { filePath: string, uploadUrl: string } => {
```

Comment:
More of a preference thing but you can also filter like this if you want to omit `undefined` values
```suggestion
  }).filter(Boolean)
```

Reply:
Both effectively filter, but the `pair is { filePath: string, uploadUrl: string }` part is pretty important, because typescript will infer the return type as not `| undefined` with it